### PR TITLE
Linting - breaking changes to internal API to prepare

### PR DESCRIPTION
**Pull request by nedtwigg**
- This PR replaces #1096

Main change is that `DirtyState` is now the primary entry-point for using a `Formatter`.

### CODE REVIEW

**Comment by nedtwigg** [comment]
@jbduncan this is the same Lint PR that you reviewed earlier, but rebased and cleaned up. We're pretty close to shipping linting *and* configuration cache. Huzzah!

File: lib/src/main/java/com/diffplug/spotless/DirtyState.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+		out.write(canonicalBytes());
+	}
+
+	/** Returns the DirtyState which corresponds to {@code isClean()}. */
+	public static DirtyState clean() {
+		return isClean;
+	}
+
+	static final DirtyState didNotConverge = new DirtyState(null);
+	static final DirtyState isClean = new DirtyState(null);
```
Comment (Goooler) [inline lib/src/main/java/com/diffplug/spotless/DirtyState.java:70]:
```suggestion
	static final DirtyState cleanState = new DirtyState(null);
```

**Review by Goooler — APPROVED** [review]

File: CHANGES.md [hunk 2]
```diff
@@ -34,6 +34,8 @@ We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (
 * **BREAKING** Remove `JarState.getMavenCoordinate(String prefix)`. ([#1945](https://github.com/diffplug/spotless/pull/1945))
 * **BREAKING** Replace `PipeStepPair` with `FenceStep`. ([#1954](https://github.com/diffplug/spotless/pull/1954))
 * **BREAKING** Fully removed `Rome`, use `Biome` instead. ([#2119](https://github.com/diffplug/spotless/pull/2119))
+* **BREAKING** Moved `PaddedCell.DirtyState` to its own top-level class with new methods.
+* **BREAKING** Removed `isClean`, `applyTo`, and `applyToAndReturnResultIfDirty` from `Formatter` because users should instead use `PaddedCell.check()`.
```
[the remark this answers is not in the document]
Comment (nedtwigg) [reply CHANGES.md]: Good call.

File: lib/src/main/java/com/diffplug/spotless/DirtyState.java (hunk 1 above)
Comment (nedtwigg) [reply lib/src/main/java/com/diffplug/spotless/DirtyState.java:70]: reasonable suggestion, but I'm sticking with `isClean` because I like the fluent style

**Review by Goooler — APPROVED** [review]

**Comment by lehmanju** [comment]
anything I can do to help this get done?

**Comment by nedtwigg** [comment]
@lehmanju sorry for the delay, almost there! There's a bit more work to be done #2149, then you should be unblocked!

**Comment by lehmanju** [comment]
@nedtwigg really appreciate the time and work you put into this project!

**Comment by nedtwigg** [comment]
Thanks! 190+ contributors, I just try to keep everybody's contributions working and unblocked :)

**Pull request merged**